openmc/mgxs/library.py

```diff
@@ -859,65 +866,80 @@ def get_xsdata(self, domain, xsdata_name, nuclide='total', xs_type='macro',
             xsdata.zaid = self._nuclides[nuclide][0]
             xsdata.awr = self._nuclides[nuclide][1]
 
+        if subdomain is None:
```

Comment:
Is there a reason to create a new `subdomain_val` variable here? Or can you simply reuse `subdomain` as so:

```python
if subdomain is None:
    subdomain = 'all'
else:
    subdomain = [subdomain]
```